forms-flow-api/src/formsflow_api/services/import_support.py

```diff
@@ -0,0 +1,343 @@
+            "process_name": form_response.get("name"),
+        }
+        mapper = FormProcessMapperService.create_mapper(mapper_data)
+        form_logs_data = {
+            "titleChanged": True,
+            "formName": form_response.get("title"),
+            "formTypeChanged": True,
+            "formType": form_response.get("type"),
+            "anonymousChanged": True,
+            "anonymous": False,
+            "formId": form_id,
+            "parentFormId": form_id,
+        }
+        FormHistoryService.create_form_logs_without_clone(data=form_logs_data)
+        # Update the form_id in the data dictionary with the new form_id
+        for auth in file_data["authorizations"][0]:
+            file_data["authorizations"][0][auth]["resourceId"] = form_id
+        # Create authorizations for the form
+        self.create_authorization(file_data["authorizations"][0])
+        self.save_process_data(
+            workflow_data, form_response.get("name"), mapper_id=mapper.id, is_new=True
+        )
+
+    def find_mapper(self, mapper_id, tenant_key=None):
+        """Find mapper."""
+        mapper = FormProcessMapper.find_form_by_id(form_process_mapper_id=mapper_id)
+        if mapper and tenant_key is not None and mapper.tenant != tenant_key:
+            raise PermissionError(BusinessErrorCode.PERMISSION_DENIED)
+        return mapper
+
+    @user_context
+    def import_form_workflow(self, request, **kwargs):  # pylint: disable=too-many-locals, too-many-statements, too-many-branches
+        """Import form/workflow."""
+        current_app.logger.info("Processing import data...")
+        request_data, file = self.validate_input_data(request)
+        schema = ImportRequestSchema()
+        input_data = schema.load(request_data)
+        import_type = input_data.get("import_type")
+        action = input_data.get("action")
+        user: UserContext = kwargs["user"]
+        tenant_key = user.tenant_key
+
+        # Check if the action is valid
+        if action not in ["validate", "import"]:
+            raise BusinessException(BusinessErrorCode.INVALID_INPUT)
+
+        if import_type == "new":
+            current_app.logger.info("Import new processing..")
+            # Validate input file type whether it is json
+            if not self.validate_file_type(file.filename, (".json",)):
+                raise BusinessException(BusinessErrorCode.INVALID_FILE_TYPE)
+            current_app.logger.info("Valid json file type.")
+            file_data = self.read_json_data(file)
+            # Vaidate file data
+            self.validate_input_json(file_data, form_workflow_schema)
+            form_json = file_data.get("forms")[0].get("content")
+            workflow_data = file_data.get("workflows")[0].get("content")
+            # Check the form name/path/title already exists
+            query_params = (
+                f"title={form_json.get('title')}&"
+                f"name={form_json.get('name')}&"

```

Comment:
If the environment is using multi tenant, and if tenantKey is not there in the path/name then we need to prepend tenantKey on search. Are we including the tenantKey on export ?

Reply:
done